### PR TITLE
Add support for Core components written in TypeScript

### PR DESCRIPTION
**Pull request by craigpalermo**
**Changes included:**

- Rewrite `Box` component and related files in TypeScript
- Update Storybook project to SB 6 for the built-in TS support. This also fixes the broken docs pages in the existing storybook.
- Delete unused `types` project

Tracking [this issue](https://github.com/storybookjs/storybook/issues/13304) related to prop table on docsite for TS components and [this issue](https://github.com/storybookjs/storybook/issues/11933) related to missing docs page props for styled components.

**Todo**

- [x] Fix error in `rush lint:fix`
- [x] Run `rush change`
- [ ] Confirm that prop types appear on Storybook Docs page for `Box`, as well as JS components

### CODE REVIEW

File: packages/core/src/Avatar/Avatar.stories.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-  .add('Elon', () => (
-    <Avatar
-      title='Not Elon Musk'
-      subtitle='totally.not.elon@musk.com'
-      src={elonJPG}
-    />
-  ))
+export default {
+  title: 'Avatar',
+};
```
Comment (craigpalermo) [inline packages/core/src/Avatar/Avatar.stories.js:8]: Add `component`

File: packages/core/src/Banner/Banner.stories.js [hunk 2]
```diff
@@ -1,719 +1,531 @@
...
-        onClose={action('closed')}
-      >
-        <Text bold italic>
-          I am a text component!
-        </Text>
-      </Banner>
-    </Box>
-  ))
+export default {
+  title: 'Banner',
```
Comment (craigpalermo) [inline packages/core/src/Banner/Banner.stories.js:6]: Add `component`

File: packages/core/src/BlockLink/BlockLink.stories.js [hunk 3]
```diff
@@ -1,51 +1,61 @@
...
-        <Text fontSize={2} bold textAlign='center'>
-          Click to go to google.com!
-        </Text>
-      </BlockLink>
-    </Flex>
-  ))
+  'https://images.unsplash.com/photo-1446776811953-b23d57bd21aa?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=aee8a50c86478d935556d865624506e4';
+
+export default {
+  title: 'BlockLink',
```
Comment (craigpalermo) [inline packages/core/src/BlockLink/BlockLink.stories.js:10]: Add `component`

File: packages/core/src/Breadcrumbs/Breadcrumbs.stories.js [hunk 4]
```diff
@@ -1,73 +1,66 @@
...
-            onClick={() =>
-              (dsRef.current.innerHTML = "What's the frequency, Kenneth?")
-            }
-          >
-            Click to change the first link via ref
-          </Button>
-        </>
-      )}
+export default {
+  title: 'Breadcrumbs',
```
Comment (craigpalermo) [inline packages/core/src/Breadcrumbs/Breadcrumbs.stories.js:8]: Add `component`

File: packages/core/src/Card/Card.stories.js [hunk 5]
```diff
@@ -1,149 +1,69 @@
...
-        color='black'
-        bg='white'
-        borderRadius='30px'
-      >
-        XLarge Shadow - 30px
-      </Card>
-    </Box>
-  ))
+export default {
+  title: 'Card',
```
Comment (craigpalermo) [inline packages/core/src/Card/Card.stories.js:5]: Add `component`

File: packages/core/src/Checkbox/Checkbox.stories.js [hunk 6]
```diff
@@ -1,172 +1,157 @@
...
+};
 
-      <Wrapper>
-        <StyledLabel htmlFor='checked_box'>
-          <Checkbox id='checked_box' defaultChecked onChange={checkAction} />
-          Checked by default
-        </StyledLabel>
-      </Wrapper>
+export default {
+  title: 'Checkbox',
```
Comment (craigpalermo) [inline packages/core/src/Checkbox/Checkbox.stories.js:32]: Add `component`

File: packages/core/src/CloseButton/CloseButton.stories.js [hunk 7]
```diff
@@ -1,28 +1,35 @@
...
-          <br />
-          <Button onClick={() => dsRef.current.focus()} mt={4}>
-            Click to focus button via ref
-          </Button>
-        </>
-      )}
-    />
-  ))
+export default {
+  title: 'CloseButton',
```
Comment (craigpalermo) [inline packages/core/src/CloseButton/CloseButton.stories.js:8]: Add `component`

File: packages/core/src/Container/Container.stories.js [hunk 8]
```diff
@@ -1,19 +1,30 @@
...
-  ))
-  .add('Input maxWidth', () => (
-    <Container maxWidth={500}>
-      <Box p={4} bg='lightGray' style={{ height: `100vh` }}>
-        Container Compnent
-      </Box>
-    </Container>
-  ))
+export default {
+  title: 'Container',
```
Comment (craigpalermo) [inline packages/core/src/Container/Container.stories.js:5]: Add `component`

File: packages/core/src/Flag/Flag.stories.js [hunk 9]
```diff
@@ -1,91 +1,114 @@
...
-          Hello Blue
-        </Flag>
-        <Flag mt={3} bg='purple'>
-          Hello Purple
-        </Flag>
-      </Card>
-    </Box>
-  ))
+export default {
+  title: 'Flag',
```
Comment (craigpalermo) [inline packages/core/src/Flag/Flag.stories.js:6]: Add `component`

File: packages/core/src/Flex/Flex.stories.js [hunk 10]
```diff
@@ -1,55 +1,69 @@
...
-      <Box width={1 / 2} p={1} color='white' bg='green'>
-        Box
-      </Box>
-    </Flex>
-  ))
+import React from 'react';
+import { Flex, Box } from '..';
+
+export default {
+  title: 'Flex',
```
Comment (craigpalermo) [inline packages/core/src/Flex/Flex.stories.js:5]: Add `component`

File: packages/core/src/FormField/FormField.stories.js [hunk 11]
```diff
@@ -1,124 +1,150 @@
...
-        placeholder='hello@example.com'
-        color='success'
-      />
-      <SuccessIcon color='success' />
-    </FormField>
-  ))
-  .add('with error Tooltip', () => (
-    <Box>
+export default {
+  title: 'FormField',
```
Comment (craigpalermo) [inline packages/core/src/FormField/FormField.stories.js:12]: Add `component`

File: packages/core/src/Hide/Hide.stories.js [hunk 12]
```diff
@@ -1,44 +1,47 @@
-import React from 'react'
-import { storiesOf } from '@storybook/react'
-import { Hide, Box, Flex } from '..'
+import React from 'react';
+import { Hide, Box, Flex } from '..';
 
-storiesOf('Hide', module).add('Hide', () => (
-  <Flex justifyContent='space-between'>
+export default {
+  title: 'Hide',
```
Comment (craigpalermo) [inline packages/core/src/Hide/Hide.stories.js:5]: Add `component`

File: packages/core/src/Hug/Hug.stories.js [hunk 13]
```diff
@@ -1,78 +1,108 @@
...
-        None: 'none',
-      })}
-    >
-      <Card p={3} bg='white' color='text'>
-        I‘m a card within a hug!
+);
+
+export default {
+  title: 'Hug',
+  decorators: [withKnobs],
```
Comment (craigpalermo) [inline packages/core/src/Hug/Hug.stories.js:24]: Add `component`

File: packages/core/src/IconButton/IconButton.stories.js [hunk 14]
```diff
@@ -1,51 +1,74 @@
...
-  ))
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { Calendar } from 'pcln-icons';
+
+import { Button, IconButton } from '..';
+import ForwardRefDemo from '../../storybook/utils/ForwardRefsDemo';
+
+export default {
+  title: 'IconButton',
```
Comment (craigpalermo) [inline packages/core/src/IconButton/IconButton.stories.js:9]: Add `component`

File: packages/core/src/IconField/IconField.stories.js [hunk 15]
```diff
@@ -1,64 +1,91 @@
...
-    <IconField>
-      <CalendarIcon color='blue' />
-      <Select>
-        <option>Choose Date</option>
-        <option>January 2019</option>
-      </Select>
-    </IconField>
-  ))
+export default {
+  title: 'IconField',
```
Comment (craigpalermo) [inline packages/core/src/IconField/IconField.stories.js:7]: Add `component`

File: packages/core/src/Link/Link.stories.js [hunk 16]
```diff
@@ -23,55 +22,70 @@ const colors = {
...
-      color={Object.keys(colors)}
-      variation={Object.keys(variations)}
-      onClick={action('Clicked button in All')}
-      m={3}
-    >
-      Link Text
-    </Cartesian>
-  ))
+export default {
+  title: 'Link',
```
Comment (craigpalermo) [inline packages/core/src/Link/Link.stories.js:28]: Add `component`

File: packages/core/src/Radio/Radio.stories.js [hunk 17]
```diff
@@ -54,40 +53,57 @@ class MockForm extends React.Component {
...
-          </Label>
-          <Button onClick={() => dsRef.current.focus()} mt={4}>
-            Click to focus radio via ref
-          </Button>
-        </>
-      )}
-    />
-  ))
+export default {
+  title: 'Radio',
```
Comment (craigpalermo) [inline packages/core/src/Radio/Radio.stories.js:61]: Add `component`

File: packages/core/src/RatingBadge/RatingBadge.stories.js [hunk 18]
```diff
@@ -1,7 +1,12 @@
...
-import { storiesOf } from '@storybook/react'
-import { RatingBadge } from '..'
+import React from 'react';
+import { RatingBadge } from '..';
 
-storiesOf('RatingBadge', module).add('default', () => (
-  <RatingBadge>9.0</RatingBadge>
-))
+export default {
+  title: 'RatingBadge',
```
Comment (craigpalermo) [inline packages/core/src/RatingBadge/RatingBadge.stories.js:5]: Add `component`

File: packages/core/src/Relative/Relative.stories.js [hunk 19]
```diff
@@ -1,31 +1,41 @@
...
+import { Absolute, Card, Flag, Image, Relative, Text } from '..';
+import { Close as CloseIcon } from 'pcln-icons';
 
-storiesOf('Relative', module)
-  .add('Around an Image and an absolutely positioned Flag', () => (
-    <Relative width={1 / 2}>
-      <Absolute top={8} left={0}>
-        <Flag>Hello Flag</Flag>
+export default {
+  title: 'Relative',
```
Comment (craigpalermo) [inline packages/core/src/Relative/Relative.stories.js:6]: Add `component`

File: packages/core/src/Select/Select.stories.js [hunk 20]
```diff
@@ -1,61 +1,78 @@
...
-      )}
-    />
-  ))
+import React from 'react';
+
+import { Select, Label, Box, Button } from '..';
+import ForwardRefDemo from '../../storybook/utils/ForwardRefsDemo';
+
+export default {
+  title: 'Select',
```
Comment (craigpalermo) [inline packages/core/src/Select/Select.stories.js:7]: Add `component`

File: packages/core/src/SrOnly/SrOnly.stories.js [hunk 21]
```diff
@@ -1,19 +1,25 @@
...
+import React from 'react';
+import { SrOnly, Text, BackgroundImage, Box } from '..';
 
 const image =
-  'https://images.unsplash.com/photo-1446776811953-b23d57bd21aa?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=aee8a50c86478d935556d865624506e4'
+  'https://images.unsplash.com/photo-1446776811953-b23d57bd21aa?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&s=aee8a50c86478d935556d865624506e4';
 
-storiesOf('SrOnly', module).add('Screenreader description of content', () => (
+export default {
+  title: 'SrOnly',
```
Comment (craigpalermo) [inline packages/core/src/SrOnly/SrOnly.stories.js:8]: Add `component`

File: packages/core/src/Tooltip/Tooltip.stories.js [hunk 22]
```diff
@@ -1,127 +1,130 @@
...
-`
+`;
 
-storiesOf('Tooltip', module)
-  .add('Without Anchors', () => (
-    <Box mt={5} width={500}>
-      <Tooltip bg='blue' color='white' top left>
-        left tooltip
+export default {
+  title: 'Tooltip',
```
Comment (craigpalermo) [inline packages/core/src/Tooltip/Tooltip.stories.js:10]: Add `component`

File: packages/core/src/Truncate/Truncate.stories.js [hunk 23]
```diff
@@ -12,12 +11,16 @@ Quod equidem non reprehendo; Facete M. Etsi ea quidem, quae adhuc dixisti, quamv
...
-  .add('Without Container', () => <Truncate width={50}>{loripsum}</Truncate>)
-  .add('With Container', () => (
-    <Box width={3 / 10}>
-      <Truncate>{loripsum}</Truncate>
-    </Box>
-  ))
+`;
+
+export default {
+  title: 'Truncate',
```
Comment (craigpalermo) [inline packages/core/src/Truncate/Truncate.stories.js:17]: Add `component`

File: packages/core/test/jest.setup.ts [hunk 24]
```diff
@@ -0,0 +1,18 @@
+import '@testing-library/jest-dom/extend-expect'
+import 'jest-styled-components'
+
+import { rendererCreateWithTheme } from './rendererCreateWithTheme'
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (craigpalermo) [inline packages/core/test/jest.setup.ts:6]: Change to file-level disable

**Comment by sdalonzo** [comment]
I have a SB6 branch as well, I don't think it warrants having open until SB gets that issue resolved.

**Comment by craigpalermo** [comment]
Blocked by the fix in #995

**Pull request closed without merging**